src/components/Grid/Grid.js

```diff
@@ -0,0 +1,24 @@
+import React from 'react'
+import { Grid } from '@mui/material'
+
+
+export const ControlledGrig = (props) => {
+    const {
+        rowSpacing,
+        columnSpacing,
+        xs,
+        arrMapping = [],
+        style,
+    } = props
+    return (
+        <Grid style={style} container rowSpacing={rowSpacing} columnSpacing={columnSpacing}>
+            {
+                arrMapping.map((item) =>(
+                    <Grid item xs={xs}>
```

Comment:
**xs** wont work accurately here. the objective is to have an **xs** on each every index from array. but here you are passing an **xs** wherein it is standalone and not dynamic. you should use **item.xs**